orchestration/orchestrator.go

```diff
@@ -47,6 +46,46 @@ type Orchestrator struct {
 	ServiceDiscovery servicediscovery.ServiceDiscovery
 	// Token is a uniqueness token for calls to AWS
 	Token string
+	// DefaultPublic disables the setting of public IPs on ENIs by default
+	DefaultPublic string
+	// DefaultSubnets sets a list of default subnets to attach ENIs
+	DefaultSubnets []string
+	// DefaultSecurityGroups sets a list of default sgs to attach to ENIs
+	DefaultSecurityGroups []string
 	// Org is the organization where this orchestration runs
 	Org string
 }
+
+type rollbackFunc func(ctx context.Context) error
+
+// rollBack executes functions from a stack of rollback functions
+func rollBack(t *[]rollbackFunc) {
+	if t == nil {
+		return
+	}
+
+	timeout, cancel := context.WithTimeout(context.Background(), 120*time.Second)
+	defer cancel()
+
+	done := make(chan string, 1)
+	go func() {
+		tasks := *t
+		log.Errorf("executing rollback of %d tasks", len(tasks))
```

Comment:
So is the rollback going to run in the background, i.e. return a response before it's finished?

Reply:
yeah a bunch of the logic needs to retry (like waiting for everything to be cleaned up before deleting the cluster) so I figured it made sense to return and do it in the background.  I don't actually think the client cares about the cleanup/rollback.  I think there's a _slight_ chance of race on the cluster since it's shared between services but otherwise I think it's safe.